src/new/sendTransaction.js

```diff
@@ -0,0 +1,94 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import EventEmitter from 'events';
+import { MIN_POLL } from '../constants';
+import transaction from '../transaction';
+import txApi from '../api/txApi';
+import txMiningApi from '../api/txMining';
+import { AddressError, OutputValueError, ConstantNotSet, MaximumNumberOutputsError, MaximumNumberInputsError } from '../errors';
+
+/**
+ * This is transaction mining class responsible for:
+ *
+ * - Submit a job to be mined;
+ * - Update mining time estimation from time to time;
+ * - Get back mining response;
+ * - Push tx to the network;
+ **/
+class SendTransaction extends EventEmitter {
+  /*
+   * data {Object} Tx data
+   */
+  constructor({
+    data=null,
+  } = {}) {
+    super();
+
+    this.data = data;
+    this.estimation = null;
+    this.jobID = null;
+  }
+
+  submitJob() {
+    // Get tx hex without parents and nonce
+    const txHex = transaction.getTxHexFromData(this.data);
+    // Send to be mined in tx mining API
+    txMiningApi.submitJob(txHex, false, true, (response) => {
+      this.estimation = response.expected_total_time;
+      this.jobID = response.job_id;
+      this.emit('job-submitted', {estimation: this.estimation, jobID: this.jobID});
+      this.handleJobStatus();
+    });
+  }
+
+  handleJobStatus() {
+    // this.estimation and MIN_POLL are in seconds
+    const poll_time = Math.max(this.estimation / 2, MIN_POLL)*1000;
+
+    setTimeout(() => {
+      txMiningApi.getJobStatus(this.jobID, (response) => {
+        if (response.status === 'done') {
+          this.data.nonce = response.nonce;
+          this.data.parents = response.parents;
+          this.emit('job-done', {jobID: this.jobID});
+          this.handlePushTx();
+        } else {
+          this.estimation = response.expected_total_time;
+          this.emit('estimation-updated', {estimation: response.expected_total_time});
+          this.handleJobStatus();
+        }
+      });
+    }, poll_time);
+  }
+
+  handlePushTx() {
+    const txHex = transaction.getTxHexFromData(this.data);
+    txApi.pushTx(txHex, false, (response) => {
```

Comment:
Need to handle the error case and probably emit a corresponding event.

Reply:
My first idea what to return only one event with the request response, so people can check the success. But thinking about it now, maybe it's not the best approach. I will emit an event for success with the sent tx data and an event for error, with the error message.

Is that it, right?

Reply:
Actually, I meant that `txApi.pushTx()` can return a promise rejection, which is not being handled.

What you're saying is that the `tx-sent` event would have both the success and error messages?

Reply:
Yes, but not anymore. I've changed that to emit different events (for success and error) and added the promise reject error event also.

Reply:
Looks good now.